TESTS WITH ERRORS 
test03_GM_Hanoi_03 (gradingtests.UnitTests)
Traceback (most recent call last):
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 108, in test03_GM_Hanoi_03
    th.makeMove(movables[0])
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_game_masters.py", line 155, in makeMove
    statement5 = Statement(['ontop', disktomove, topofnewpegdisk])
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\logical_classes.py", line 181, in __init__
    self.terms = [t if isinstance(t, Term) else Term(t) for t in statement_list[1:]]
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\logical_classes.py", line 181, in <listcomp>
    self.terms = [t if isinstance(t, Term) else Term(t) for t in statement_list[1:]]
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\logical_classes.py", line 242, in __init__
    self.term = term if is_var_or_const else (Variable(term) if is_var(term) else Constant(term))
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\util.py", line 15, in is_var
    return var[0] == "?"
IndexError: string index out of range

test04_GM_Hanoi_04 (gradingtests.UnitTests)
Traceback (most recent call last):
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 123, in test04_GM_Hanoi_04
    th.makeMove(movables[0])
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_game_masters.py", line 155, in makeMove
    statement5 = Statement(['ontop', disktomove, topofnewpegdisk])
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\logical_classes.py", line 181, in __init__
    self.terms = [t if isinstance(t, Term) else Term(t) for t in statement_list[1:]]
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\logical_classes.py", line 181, in <listcomp>
    self.terms = [t if isinstance(t, Term) else Term(t) for t in statement_list[1:]]
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\logical_classes.py", line 242, in __init__
    self.term = term if is_var_or_const else (Variable(term) if is_var(term) else Constant(term))
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\util.py", line 15, in is_var
    return var[0] == "?"
IndexError: string index out of range

test10_DFS_Hanoi_02 (gradingtests.UnitTests)
Traceback (most recent call last):
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 246, in test10_DFS_Hanoi_02
    [20, ((2,5,), (4,), (1, 3))],
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 52, in runPlayXSteps
    results = self.pool.apply_async(self.playXSteps, [solver, plays]).get(timeout)
  File "C:\Program Files\Python37\lib\multiprocessing\pool.py", line 683, in get
    raise self._value
  File "C:\Program Files\Python37\lib\multiprocessing\pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 27, in playXSteps
    solver.solveOneStep()
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_uninformed_solvers.py", line 76, in solveOneStep
    self.gm.reverseMove(self.moverecords.pop())
IndexError: pop from empty list

test11_DFS_Hanoi_03 (gradingtests.UnitTests)
Traceback (most recent call last):
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 260, in test11_DFS_Hanoi_03
    self.runSolve(solver)
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 68, in runSolve
    self.pool.apply_async(self.solve, [solver,]).get(timeout)
  File "C:\Program Files\Python37\lib\multiprocessing\pool.py", line 683, in get
    raise self._value
  File "C:\Program Files\Python37\lib\multiprocessing\pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 40, in solve
    solver.solve()
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\solver.py", line 45, in solve
    while not self.solveOneStep():
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_uninformed_solvers.py", line 67, in solveOneStep
    self.gm.reverseMove(self.moverecords.pop())
IndexError: pop from empty list

test13_BFS_Hanoi_02 (gradingtests.UnitTests)
Traceback (most recent call last):
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 296, in test13_BFS_Hanoi_02
    [23, ((5,), (4,), (1, 2, 3))],
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 52, in runPlayXSteps
    results = self.pool.apply_async(self.playXSteps, [solver, plays]).get(timeout)
  File "C:\Program Files\Python37\lib\multiprocessing\pool.py", line 683, in get
    raise self._value
  File "C:\Program Files\Python37\lib\multiprocessing\pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 27, in playXSteps
    solver.solveOneStep()
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_uninformed_solvers.py", line 124, in solveOneStep
    self.gm.makeMove(move) # move to that child
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_game_masters.py", line 155, in makeMove
    statement5 = Statement(['ontop', disktomove, topofnewpegdisk])
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\logical_classes.py", line 181, in __init__
    self.terms = [t if isinstance(t, Term) else Term(t) for t in statement_list[1:]]
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\logical_classes.py", line 181, in <listcomp>
    self.terms = [t if isinstance(t, Term) else Term(t) for t in statement_list[1:]]
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\logical_classes.py", line 242, in __init__
    self.term = term if is_var_or_const else (Variable(term) if is_var(term) else Constant(term))
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\util.py", line 15, in is_var
    return var[0] == "?"
IndexError: string index out of range

test14_BFS_Hanoi_03 (gradingtests.UnitTests)
Traceback (most recent call last):
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 314, in test14_BFS_Hanoi_03
    [23, ((1, 4, 5), (), (2, 3))],
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 52, in runPlayXSteps
    results = self.pool.apply_async(self.playXSteps, [solver, plays]).get(timeout)
  File "C:\Program Files\Python37\lib\multiprocessing\pool.py", line 683, in get
    raise self._value
  File "C:\Program Files\Python37\lib\multiprocessing\pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 27, in playXSteps
    solver.solveOneStep()
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_uninformed_solvers.py", line 124, in solveOneStep
    self.gm.makeMove(move) # move to that child
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_game_masters.py", line 155, in makeMove
    statement5 = Statement(['ontop', disktomove, topofnewpegdisk])
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\logical_classes.py", line 181, in __init__
    self.terms = [t if isinstance(t, Term) else Term(t) for t in statement_list[1:]]
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\logical_classes.py", line 181, in <listcomp>
    self.terms = [t if isinstance(t, Term) else Term(t) for t in statement_list[1:]]
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\logical_classes.py", line 242, in __init__
    self.term = term if is_var_or_const else (Variable(term) if is_var(term) else Constant(term))
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\util.py", line 15, in is_var
    return var[0] == "?"
IndexError: string index out of range

test18_BFS_8Puzzle_01 (gradingtests.UnitTests)
Traceback (most recent call last):
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 390, in test18_BFS_8Puzzle_01
    [21, ((6, 5, 4), (1, -1, 8), (7, 3, 2))],
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 52, in runPlayXSteps
    results = self.pool.apply_async(self.playXSteps, [solver, plays]).get(timeout)
  File "C:\Program Files\Python37\lib\multiprocessing\pool.py", line 683, in get
    raise self._value
  File "C:\Program Files\Python37\lib\multiprocessing\pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 27, in playXSteps
    solver.solveOneStep()
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_uninformed_solvers.py", line 148, in solveOneStep
    self.gm.makeMove(step)
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_game_masters.py", line 283, in makeMove
    newx = terms[3].__str__()
IndexError: list index out of range

test19_BFS_8Puzzle_02 (gradingtests.UnitTests)
Traceback (most recent call last):
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 409, in test19_BFS_8Puzzle_02
    [6, ((5, 4, 8), (6, 1, 2), (7, 3, -1))],
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 52, in runPlayXSteps
    results = self.pool.apply_async(self.playXSteps, [solver, plays]).get(timeout)
  File "C:\Program Files\Python37\lib\multiprocessing\pool.py", line 683, in get
    raise self._value
  File "C:\Program Files\Python37\lib\multiprocessing\pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 27, in playXSteps
    solver.solveOneStep()
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_uninformed_solvers.py", line 148, in solveOneStep
    self.gm.makeMove(step)
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_game_masters.py", line 283, in makeMove
    newx = terms[3].__str__()
IndexError: list index out of range

test20_BFS_8Puzzle_03 (gradingtests.UnitTests)
Traceback (most recent call last):
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 428, in test20_BFS_8Puzzle_03
    [7, ((1, 2, -1), (8, 4, 3), (7, 6, 5))],
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 52, in runPlayXSteps
    results = self.pool.apply_async(self.playXSteps, [solver, plays]).get(timeout)
  File "C:\Program Files\Python37\lib\multiprocessing\pool.py", line 683, in get
    raise self._value
  File "C:\Program Files\Python37\lib\multiprocessing\pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 27, in playXSteps
    solver.solveOneStep()
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_uninformed_solvers.py", line 148, in solveOneStep
    self.gm.makeMove(step)
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\student_code_game_masters.py", line 283, in makeMove
    newx = terms[3].__str__()
IndexError: list index out of range

-------------------------------
TESTS FAILED  
test08_GM_8Puzzle_04 (gradingtests.UnitTests)
Traceback (most recent call last):
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 200, in test08_GM_8Puzzle_04
    self.assertEqual(p8.getGameState(), ((1,-1,3,),(8,2,4),(7,6,5)))
AssertionError: Tuples differ: ((1, 2, 3), (8, -1, 4), (7, 6, 5)) != ((1, -1, 3), (8, 2, 4), (7, 6, 5))

First differing element 0:
(1, 2, 3)
(1, -1, 3)

- ((1, 2, 3), (8, -1, 4), (7, 6, 5))
?      ^          ^^

+ ((1, -1, 3), (8, 2, 4), (7, 6, 5))
?      ^^          ^


test16_DFS_8Puzzle_02 (gradingtests.UnitTests)
Traceback (most recent call last):
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 352, in test16_DFS_8Puzzle_02
    [30, ((8, 1, -1), (7, 2, 3), (6, 5, 4))],
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 55, in runPlayXSteps
    self.assertEqual(results[index], expected)
AssertionError: Tuples differ: ((1, 2, 3), (8, 4, -1), (7, 6, 5)) != ((8, 3, 4), (7, -1, 1), (6, 2, 5))

First differing element 0:
(1, 2, 3)
(8, 3, 4)

- ((1, 2, 3), (8, 4, -1), (7, 6, 5))
+ ((8, 3, 4), (7, -1, 1), (6, 2, 5))

test17_DFS_8Puzzle_03 (gradingtests.UnitTests)
Traceback (most recent call last):
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 371, in test17_DFS_8Puzzle_03
    [8, ((8, 3, 4), (2, -1, 1), (7, 6, 5))],
  File "C:\Users\kille\academics\automatedgradingapp\app\testing\uninformed-solvers\gradingtests.py", line 55, in runPlayXSteps
    self.assertEqual(results[index], expected)
AssertionError: Tuples differ: ((-1, 2, 3), (1, 8, 4), (7, 6, 5)) != ((8, 1, 3), (-1, 2, 4), (7, 6, 5))

First differing element 0:
(-1, 2, 3)
(8, 1, 3)

- ((-1, 2, 3), (1, 8, 4), (7, 6, 5))
+ ((8, 1, 3), (-1, 2, 4), (7, 6, 5))

-------------------------------
TESTS SKIPPED 
-------------------------------
